force-app/main/default/classes/ClearAllDoctorsFieldBatch.cls

```diff
@@ -0,0 +1,33 @@
+public with sharing class ClearAllDoctorsFieldBatch implements Database.Batchable<SObject>, Database.Stateful{
+    private static Integer recordsProcessed = 0;
+    public Database.QueryLocator start (Database.BatchableContext context){
+    return Database.getQueryLocator('SELECT Id, Best_doctor_of_the_month__c, Payments_amount__c FROM Contact');
```

Comment:
You can do a more accurate query to not query doctors with Payments_amount__c = 0